### PR TITLE
feat: export data to customerio

### PR DESCRIPTION
**Pull request by idoshamun**
Update user profiles in customerio and delete them when needed.

Sidenote:  I refactored relevant workers to typed pubsub 💪

### CODE REVIEW

File: src/workers/userUpdatedCio.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+function camelCaseToSnakeCase(
+  obj: Record<string, unknown>,
+): Record<string, unknown> {
+  const snakeObj: Record<string, unknown> = {};
+  for (const key in obj) {
+    snakeObj[key.replace(/[A-Z]/g, (letter) => `_${letter.toLowerCase()}`)] =
+      obj[key];
+  }
+  return snakeObj;
+}
```
Comment (omBratteng) [inline src/workers/userUpdatedCio.ts:15]: make this a global util?

File: src/workers/userUpdatedCio.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  'bio',
+  'devcardEligible',
+  'readme',
+  'readmeHtml',
+  'infoConfirmed',
+  'profileConfirmed',
+  'notificationEmail',
+];
+
+const worker: TypedWorker<'user-updated'> = {
```
Comment (omBratteng) [inline src/workers/userUpdatedCio.ts:28]: Just for updates? Or does it also run on user created?

File: src/workers/userUpdatedCio.ts (hunk 1 above)
Comment (idoshamun) [reply src/workers/userUpdatedCio.ts:15]: yes!

File: src/workers/userUpdatedCio.ts (hunk 2 above)
Comment (idoshamun) [reply src/workers/userUpdatedCio.ts:28]: originally i thought that on creation a user can't be confirmed but i think with email they can. i'll double check

File: src/cio.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { TrackClient } from 'customerio-node';
+
+export const cio = new TrackClient(
+  process.env.CIO_SITE_ID,
+  process.env.CIO_API_KEY,
+);
```
Comment (capJavert) [inline src/cio.ts:6]: Just to make sure, will this work locally where we don't have credentials in env?

File: src/workers/userUpdatedCio.ts (hunk 1 above)
Comment (capJavert) [reply src/workers/userUpdatedCio.ts:15]: One edge case with implementation:
```js
camelCaseToSnakeCase({
    SomeProp: 'test'
})
```

outputs:
```
{
    _some_prop: 'test'
}
```

Comment (capJavert) [reply src/workers/userUpdatedCio.ts:15]: Since we already have lodash maybe use `_.snakeCase`?

File: src/workers/userUpdatedCio.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+const OMIT_FIELDS: (keyof User)[] = [
+  'id',
+  'bio',
+  'devcardEligible',
+  'readme',
+  'readmeHtml',
+  'infoConfirmed',
+  'profileConfirmed',
+  'notificationEmail',
+];
```
Comment (capJavert) [inline src/workers/userUpdatedCio.ts:26]: Does it make sense to do it other way around, specify fields we want to send? I am thinking we add some field to `User` eg. `flags` and then forget to update it we accidentally send it to customerio.

**Review by capJavert — APPROVED** [review]
Looks good, some minor comments.

File: src/workers/userUpdatedCio.ts (hunk 4 above)
Comment (idoshamun) [reply src/workers/userUpdatedCio.ts:26]: I rather have them sent mistakenly than not send. So imagine we add experience level (like we did) and totally forget about it. Then we have to import historical data and fix the code, more like risk management

File: src/cio.ts (hunk 3 above)
Comment (idoshamun) [reply src/cio.ts:6]: i'll double check but it shouldn't break, probably will just throw an error if you try to actually run it, but not on init

File: src/workers/userUpdatedCio.ts (hunk 1 above)
Comment (idoshamun) [reply src/workers/userUpdatedCio.ts:15]: Good idea!

File: src/cio.ts (hunk 3 above)
Comment (capJavert) [reply src/cio.ts:9]: It is gonna fire when we create user locally right, and then worker will try to retry indefinitely. I think we had similar thing with sendgrid and then we just added NODE_ENV check to avoid running worker code where it is required in development.

File: __tests__/workers/userCreatedCio.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  ...(jest.requireActual('../../src/cio') as Record<string, unknown>),
+  cio: { identify: jest.fn() },
+}));
+
+beforeEach(async () => {
+  jest.clearAllMocks();
+  nock.cleanAll();
+});
+
+describe('userCreatedCio', () => {
```
Comment (capJavert) [inline __tests__/workers/userCreatedCio.ts:19]: If you can add this kind of test https://github.com/dailydotdev/daily-api/blob/45364867794974b43f4713e3fcf362da2eb27f65/__tests__/workers/commentDownvotedRep.ts#L53-L59 for new CIO workers, just to make sure they are actually registered in `workers/index`.

**Review by capJavert — APPROVED** [review]

File: src/workers/userUpdatedCio.ts (hunk 4 above)
Comment (capJavert) [reply src/workers/userUpdatedCio.ts]: 🤝

File: __tests__/workers/userCreatedCio.ts (hunk 5 above)
Comment (idoshamun) [reply __tests__/workers/userCreatedCio.ts:19]: i missed both of your comments. i'll fix that in a follow up PR
